otwcode/otwarchive
# AO3-6714 Fix duplicate tag set ownership on create

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6714

## Purpose

This PR is based on https://github.com/otwcode/otwarchive/pull/4863

This PR implements point 1. of the "What should happen" section of AO3-6714. 

This PR prevents a `OwnedTagSet` being created with duplicate `TagSetOwnership` relationships. This happens when the currently logged in user explicitly adds themselves as an owner via the UI.

### Notes

As this PR blocks creating tag sets with duplicate owners in the first place, the test case added in #4863 cannot be maintained without directly constructing an invalid state, with which to continue testing. I didn't do this; please let me know if you think it's required.

## Testing Instructions

Manually follow the recreation steps in the issue. Verify that when adding your username as an explicit owner, the created tag set only lists the currently logged in username as a maintainer once.

## References

N/A

## Credit

> What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?

Tom Milligan (they/them)
